### PR TITLE
Display report preview using new reportAction data source

### PR DESCRIPTION
**Pull request by cristipaval**
### Details

Deprecates the old usages of IOUPreview, and introduces the new ReportPreview component. We show this beneath IOUPreviews, and its position changes so that it's always beneath the most recent preview. 
 

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/278890

### Tests

**_Note: Please test with new users, as stale data will likely cause bugs_**

**1) Test report creation and new component**
- Create a new 1:1 chat between userA and userB
- Create a Request of any amount from userA to userB
- In the ChatReport, both users should see 'UserB owes £1.23' beneath the IOUPreview
- A new IOUReport should be created to hold the request, it should appear in the LHN
- Within this report, the recipient should see a 'Pay' button beneath the 'User owes...' message

**Additional Requests, and descriptions**
- Back in the chat report, create another request from userA to userB, this time with a description
- The new IOUPreview should include the description
- The 'owes £1.23' message should only display beneath the most recent IOUPreview only
- When userB taps the payButton, it should take them to the IOUDetailsPage

**Test settling up**
- UserB should settle the IOU (use Settled elsewhere as you won't have a bank set up for this new user)
- Within the IOUreport, both users should see a 'settled up $153.84 elsewhere' message
- The pay button should disappear from the 'Settled up'
- Within the ChatReport, the settled up message should show a chevron
- When tapped, it should take you back to the IOUReport

### Offline tests

Run above test, coming online to send and receive the rerquest/settle updates. We will likely handle any issues here as follow-up issues

### QA Steps

Run above tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-05-17 at 10 52 30
Screenshot 2023-05-17 at 10 52 25

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot_20230517_113128

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot_20230517_113128

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-05-17 at 11 20 31

</details>

<details>
<summary>iOS</summary>

Simulator Screen Shot - iPhone 14 - 2023-05-17 at 12 00 36

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by Julesssss** [comment]
Today I merged main again to resolve new conflicts.

I also started fixing the IOU tests. One remaining issue is that we are not clearing the reportAction's pendingAction field like we do for the other reportActions:

 
Screenshot 2023-05-15 at 15 48 52
Screenshot 2023-05-15 at 15 39 56

**Comment by Julesssss** [comment]
OKAY, for anyone taking on this PR, here's what you need to know: 

- We've been blocked from testing due to many conflicting merges and blocking PRs, bugs could well have been introduced
- The PR is blocked by [this App PR](https://github.com/Expensify/App/pull/18604). Once that is merged, we must merge pull main and merge it into this branch
- The PR must be built against [this Web-E PR](https://github.com/Expensify/Web-Expensify/pull/37438)
- We don't yet force the Avatar to show against this report action, this can follow later on

**Comment by luacmartins** [comment]
Ok, this PR should be ready. Tests well against the [Web-E](https://github.com/Expensify/Web-Expensify/pull/37438) and [Auth](https://github.com/Expensify/Auth/pull/7848) branches

**Comment by s77rt** [comment]
So I guess I have to review code and defer testing till PRs are deployed to staging, right?

**Comment by luacmartins** [comment]
Correct!

**Comment by s77rt** [comment]
Alright. I will review asap (it's midnight here)

**Comment by JmillsExpensify** [comment]
@s77rt Are you back online yet? This is a top PR we're looking to test and cherry pick to staging.

**Comment by mountiny** [comment]
Requested money for second time and the comment is there twice in the preview on iou report 
image

File: src/components/ReportActionItem/IOUPreview.js
Comment (mountiny) [inline src/components/ReportActionItem/IOUPreview.js:227]: Lets remove this, we already add comment [here](https://github.com/Expensify/App/blob/d678e0e37b1e0b5f478002b9fcf68ac57c95f54c/src/components/ReportActionItem/IOUPreview.js#L233) 
```suggestion
```

**Review by mountiny — DISMISSED** [review]

**Comment by Gonals** [comment]
> Note: Please test with new users, as stale data will likely cause bugs

What are we going to do with existing stale data?

**Comment by Julesssss** [comment]
> What are we going to do with existing stale data?

We have a BulkCQ ready to go: https://github.com/Expensify/Expensify/issues/282420

**Review by Julesssss — APPROVED** [review]
Approving @luacmartins's changes

**Review by Julesssss — APPROVED** [review]
Approving @luacmartins's changes

**Review by Gonals — APPROVED** [review]
This tests pretty well! Besides the chaos that is the LHN, with all the threads, I like it!

**Comment by s77rt** [comment]
@cristipaval Is the testing steps outdated or this is a bug? The IOUPreview is not shown in chat report it's only shown in iou report.

Screenshot from 2023-05-17 11-52-45

**Review by mountiny — APPROVED** [review]
Great job everyone, tests well

**Comment by mountiny** [comment]
@s77rt the IOU preview should be only in the IOU report.

The report preview will be in the chat

**Comment by s77rt** [comment]
Can we hold merge for about 1h? I will take a quick look in the code :grin:

**Comment by mountiny** [comment]
image


1. In workspace chat, request money
2. the report preview is not added to the workspace chat but to the expense report

File: src/components/ReportActionItem/ReportPreview.js
Comment (s77rt) [inline src/components/ReportActionItem/ReportPreview.js:1]: Minor: We have an unnecessary `mt4`

File: src/libs/actions/IOU.js
Comment (s77rt) [inline src/libs/actions/IOU.js:93]: Hmm, I'm not sure about this. Does it align with the deign docs? We are updating the `created` value of the action. Can we update another timestamp field?

File: src/libs/actions/IOU.js
Comment (s77rt) [inline src/libs/actions/IOU.js:236]: This should be wrapped in a condition `if !isNewReportPreviewAction`.

File: src/libs/ReportUtils.js
Comment (s77rt) [inline src/libs/ReportUtils.js:1430]: Missing jsdoc. (Why lint is not failing here?)

**Review by s77rt — COMMENTED** [review]
Overall code looks code. I have just few minor notes.

**Pull request merged**
